src/main/java/duke/list/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package duke.list;
+
+import duke.exceptions.IllegalOperation;
+import duke.ui.MessageBubble;
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    public static int MAX_LIST_ITEMS = 1000;
+    private final ArrayList<Task> items = new ArrayList<>();
+
+    /**
+     * Convenience constructor
+     */
+    public TaskList() {
+    }
+
+    /**
+     * Add the specified Task to the end of the TaskList.
+     * A message will be print out to show the result.
+     *
+     * @param task the task to be added
+     * @throws IllegalOperation if the number of tasks >= MAX_LIST_ITEMS
+     */
+    public void addItem(Task task) throws IllegalOperation {
+        addItem(task, true);
+    }
+
+    /**
+     * Add the specified Task to the end of the TaskList.
+     * A message will be print out to show the result if showMessage == true.
+     *
+     * @param task the task to be added
+     * @param showMessage if the message should be printed
+     * @throws IllegalOperation if the number of tasks >= MAX_LIST_ITEMS
+     */
+    public void addItem(Task task, boolean showMessage) throws IllegalOperation {
```

Comment:
What is the purpose of showMessage?

Might want to consider extracting into a separate method/class.